### PR TITLE
Add ha-alert component

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Extracts the warning part of the energy validation that was added in #9802 to a new `ha-alert` component we can reuse.

Screenshot from 2021-08-24 10-38-40

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-alert.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  public render() {
+    return html`
+      <div
+        class=${classMap({
+          "issue-type": true,
+          rtl: this.rtl,
+          info: this.alertType === "info",
+          warning: this.alertType === "warning",
+          error: this.alertType === "error",
+          success: this.alertType === "success",
```
Comment (balloob) [inline src/components/ha-alert.ts:51]: Maybe?
```suggestion
          [this.alertType]: true
```

File: src/components/ha-alert.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        })}
+      >
+        <div
+          class=${classMap({
+            icon: true,
+            info: this.alertType === "info",
+            warning: this.alertType === "warning",
+            error: this.alertType === "error",
+            success: this.alertType === "success",
```
Comment (balloob) [inline src/components/ha-alert.ts:60]: No need to do this again, you can select based on parent. .info > .icon

File: src/components/ha-alert.ts [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+          })}
+        >
+          <ha-svg-icon .path=${ALERT_ICONS[this.alertType]}></ha-svg-icon>
+        </div>
+        <div
+          class=${classMap({
+            content: true,
+            rtl: this.rtl,
```
Comment (balloob) [inline src/components/ha-alert.ts:68]: same.

File: src/components/ha-alert.ts [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+          })}
+        >
+          <div
+            class=${classMap({
+              "main-content": true,
+              no_title: !this.title,
```
Comment (balloob) [inline src/components/ha-alert.ts:74]: Let's not mix kebab and sanke case

File: src/components/ha-alert.ts [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+            })}
+          >
+            ${this.title ? html`<div class="title">${this.title}</div>` : ""}
+            <slot></slot>
+          </div>
+          <div class="action">
+            ${this.actionText
+              ? html`<mwc-button
+                  @click=${this._action_clicked}
+                  .aria-label=${this.actionText}
```
Comment (balloob) [inline src/components/ha-alert.ts:84]: we can set `.label` and mwc-button will set aria label too.

File: src/components/ha-alert.ts (hunk 1 above)
Comment (ludeeus) [reply src/components/ha-alert.ts:51]: perfect 👌

**Review by balloob — APPROVED** [review]

**Comment by balloob** [comment]
Before we merge, could you rewrite the energy validation to use this new component ?

**Comment by ludeeus** [comment]
Sure :+1: Didn't want to do that here to keep it smal

**Comment by balloob** [comment]
You're right, let's do in new PR.

**Pull request merged**